File: js/task1/S1_VAR_10.js
function inspect(value) {
  return {
      type: value === null ? "null" : typeof value,
      isArray: Array.isArray(value),
      isNull: value === null,
      isNaN: Number.isNaN(value)
  };
}

const testValues = [
  null,
  undefined,
  42,
  "hello",
  true,
  {},
  [],
  () => {},
  NaN,
  Infinity
];

testValues.forEach(value => {
  console.log(value, JSON.stringify(inspect(value)));
});

File: python/task1/S1_VAR_10.py
import math

def inspect(value):
    return {
        "type": "null" if value is None else type(value).__name__,
        "isArray": isinstance(value, list),
        "isNull": value is None,
        "isNaN": isinstance(value, float) and math.isnan(value)
    }

testValues = [
    None,
    None,
    42,
    "hello",
    True,
    {},
    [],
    lambda: None,
    float('nan'),
    float('inf')
]

for value in testValues:
    print(value, inspect(value))
